cmd/vc-event-router/main.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/connection"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/processor"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/stream"
+)
+
+// TODO: add commit/build variables
+
+func main() {
+	var config string
+	var verbose bool
+	var err error
+
+	flag.StringVar(&config, "config", "/etc/vc-event-router/config", "path to configuration file for stream source and processor connections")
+	flag.BoolVar(&verbose, "verbose", false, "print event handling information")
+	flag.Parse()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	f, err := os.Open(config)
+	if err != nil {
+		log.Fatalf("could not open configuration file: %v", err)
+	}
+	cfg, err := connection.Parse(f)
+	if err != nil {
+		log.Fatalf("could not parse configuration file: %v", err)
+	}
+
+	var (
+		streamer stream.Streamer
+		proc     processor.Processor
+	)
+
+	for _, c := range *cfg {
+		switch c.Type {
+		case "stream":
+			switch c.Provider {
+			case "vmware_vcenter":
+				streamer, err = stream.NewVCenterStream(ctx, c)
+				if err != nil {
+					log.Fatalf("could not connect to vCenter: %v", err)
+				}
+				log.Printf("connected to vCenter %s", c.Address)
+
+			default:
+				log.Fatalf("unsupported stream provider: %s", c.Provider)
+			}
+
+		case "processor":
+			switch c.Provider {
+			case "openfaas":
+				var async bool
+				if c.Options["async"] == "true" {
+					async = true
+				}
+				proc, err = processor.NewOpenFaaSProcessor(c, streamer.Source(), verbose)
+				if err != nil {
+					log.Fatalf("could not connect to OpenFaaS: %v", err)
+				}
+				log.Printf("connected to OpenFaaS gateway %s (async mode: %v)", c.Address, async)
+			case "aws_event_bridge":
+				proc, err = processor.NewAWSEventBridgeProcessor(c, streamer.Source(), verbose)
+				if err != nil {
+					log.Fatalf("could not connect to AWS EventBridge: %v", err)
+				}
+				log.Printf("connected to AWS EventBridge") // TODO: add more connection information
+			default:
+				log.Fatalf("unsupported processor provider: %s", c.Provider)
+			}
+
+		default:
+			log.Fatalf("invalid type specified: %s", c.Type)
+		}
+	}
+
+	// handle OS signals gracefully
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGTERM, os.Interrupt)
+	go func() {
+		s := <-sigCh
```

Comment:
I'm on the fence about this because `s` usually is short for `string`, but in this case, there's no confusion that it represents what comes from `sigCh`.